src/components/interpolatedColorLegend.ts

```diff
@@ -146,12 +200,10 @@ export module Component {
       if (this._isVertical()) {
         var longestWidth = _Util.Methods.max(labelWidths, 0);
         desiredWidth = this._padding + textHeight + this._padding + longestWidth + this._padding;
-        desiredHeight = this._padding + numSwatches * textHeight + this._padding;
+        desiredHeight = offeredHeight;
```

Comment:
It should only ask for all this space if it's set to expand. Your current implementation breaks `xAlign()`/`yAlign()`:
http://jsfiddle.net/9ortv6ht/